docs/devupdate.md

```diff
@@ -326,6 +326,16 @@ If you are using this functionality in older versions of Moodle you are advised
 
 :::
 
+<Since version="4.4" issueNumber="MDL-79128" />
+
+The maximum retry limit for an Ad-hoc task is **12** by default.
+
+You can set the number of retries or the remaining available attempts and get the remaining available attempts for an Ad-hoc task.
+
+It is also possible to disable the retry mechanism for an Ad-hoc task, that task will become a no-retry task.
+
+See the [detailed documentation](./apis/subsystems/task/adhoc#set-the-retry-limit-for-a-task) on how to use these new APIs.
+
```

Comment:
The thing we should mention in the developer updates documentaiotn is that we have introduced a new API to do this and changed the behaviour.

```suggestion
<Since version="4.4" issueNumber="MDL-79128" />

The behaviour of the task runner has been changed to limit the number of retries allowed before a failing _adhoc_ task is removed. Scheduled tasks are not affected.

The default value for the maximum retry limit on an Ad-hoc task is **12**.

Retries can be entirely disabled by passing a value of {{0}}, for example:

```php
$task = new example_task();
// This task wil not be retried at all.
$task->set_attempts_available(0);
\core\task\manager::queue_task($task);
```

See the [detailed documentation](./apis/subsystems/task/adhoc#set-the-retry-limit-for-a-task) on how to use these new APIs.
```